mongo.go

```diff
@@ -64,14 +47,28 @@ func newItem(i *mongoItem) *resource.Item {
 	}
 }
 
+// Handler handles resource storage in a MongoDB collection.
+type Handler func(ctx context.Context) (*mgo.Collection, error)
+
+// NewHandler creates an new mongo handler
+func NewHandler(s *mgo.Session, db, collection string) Handler {
+	return func(ctx context.Context) (*mgo.Collection, error) {
+		return s.DB(db).C(collection), nil
```

Comment:
Let's store `s.DB(db).C(collection)` in `c` outside of the function and just return `c` here.